jupyter/src/incremental_learning/job.py

```diff
@@ -577,8 +577,27 @@ def update(dataset_name: str,
         config['analysis']['parameters']['force_accept_incremental_training'] = True
 
     for name, value in original_job.get_hyperparameters().items():
-        if name not in ['retrained_tree_eta', 'tree_topology_change_penalty']:
+        if name not in ['retrained_tree_eta', 'tree_topology_change_penalty', 'soft_tree_depth_limit', 'gamma', 'alpha'] \
+            and name not in config['analysis']['parameters'].keys():
             config['analysis']['parameters'][name] = value
+        if name in ['gamma', 'alpha']:
```

Comment:
I'm not sure this apply to single step case where you have a small summary of the original data set and update with some new data. I think it would be good to factor out this logic into something like `multistep_overrides` and have some parameter to opt in to this behaviour.

Reply:
Indeed, I was also thinking along those lines. Added `hyperparameter_overrides` as a optional parameter.